include/up-cpp/client/usubscription/SubscriptionClient.h

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_CLIENT_SubscriptionClient_H
+#define UP_CPP_CLIENT_SubscriptionClient_H
+
+#include <up-cpp/communication/Subscriber.h>
+#include <uprotocol/v1/umessage.pb.h>
+
+#include <optional>
+#include <uprotocol/core/usubscription/v3/usubscription.proto>
+#include <utility>
+
+namespace uprotocol::client::usubscription {
+/// @brief Interface for uEntities to create subscriptions.
+///
+/// Like all L3 client APIs, the SubscriptionClient is a wrapper on top of the
+/// L2 Communication APIs and USubscription service.
+struct SubscriptionClient {
```

Comment:
I'm tempted to call this `Subscriber` or `Consumer` since `client` is already in the namespace. Not 100% sure there yet. What do you think?

Reply:
Subscriber may cause confusion between L2 Subscriber just in context of writing and talking. 
Might be better to call it Consumer unless Consumer might be used elsewhere too